src/vs/workbench/contrib/chat/browser/chatListRenderer.ts

```diff
@@ -833,6 +833,11 @@ export class ChatListItemRenderer extends Disposable implements ITreeRenderer<Ch
 	}
 
 	private renderProgressMessage(progress: IChatProgressMessage, showSpinner: boolean): IMarkdownRenderResult {
+		if (showSpinner) {
+			// this step is in progress, communicate it to SR users
+			console.log(progress.content.value);
+			alert(progress.content.value);
```

Comment:
This will just read the progress message, right, should there be some context that says this is a progress step?

Also, I'm not sure whether this is the right place to do this. If you scroll up while a workspace message is in progress, then scroll back down, it will read the progress update a second time.

But also, a place that updates the model might not be the right place for it, because maybe you only want this to happen if the chat is actually active on screen.

Reply:
`This will just read the progress message, right, should there be some context that says this is a progress step?`

This is already verbose, we don't need to add more words IMO. I've discussed this with Roberto.

> But also, a place that updates the model might not be the right place for it, because maybe you only want this to happen if the chat is actually active on screen.

Do you have a suggestion?

Reply:
> This is already verbose, we don't need to add more words IMO. I've discussed this with Roberto.

Sure, just remember it's uncontrolled content coming from arbitrary extensions, not just what you see in vscode-copilot.

> Do you have a suggestion?

I really don't know, it depends on in what scenario exactly you want to alert.

Reply:
In my testing, it works well